File: front-end/src/types/Course.ts
export type CourseRequest = {
  name: string;
  language: string;
  price: number;
  description: string;
  imageUrl: string;
};

export type CourseResponse = {
  id: number;
  name: string;
  startDate: string;
  endDate: string;
  imageUrl: string;
  price: number;
  description: string;
  language: string;
  studyTime: string;
};

export type Schedules = {
  id: string;
  name: string;
  description: string;
  startDate: string;
  endDate: string;
  numberOfSessions: number;
};

type Semesters = {
  id: string;
  name: string;
  description: string;
  schedules: Schedules[];
};

export type CourseResponseDetails = {
  id: number;
  name: string;
  startDate: string;
  endDate: string;
  price: number;
  description: string;
  language: string;
  semesters: Semesters[];
};
export type CourseResponseSelect = {
  id: number;
  name: string;
};

File: api/src/main/java/com/example/api/entity/Course.java
package com.example.api.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Table(name = "Course")
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "course_id")
    private Long id;

    private String name;
    private Date startDate;
    private Date endDate;
    private Integer price;
    private String imageUrl;
    private String description;
    private String language;
    private String studyTime;
    @OneToMany(mappedBy = "course")
    private Set<Semester> semesters;
}
